library/catalog/models.py

```diff
@@ -42,25 +43,35 @@ class MongoUser(Document):
     wishlist = ListField(default=[])
     preference = EmbeddedDocumentField(Preference.__name__, default=Preference())
 
-    def set_password(self, password):
-
-        self.django_password = make_password(password)
-        self.password_hash = generate_password_hash(password)
+    def generate_passwords(self, password):
+        return generate_password_hash(password), make_password(password)
 
     def check_password(self, password):
-        return check_password_hash(self.password_hash, password)
+        return check_password_hash(self.password, password)
 
     def save(self):
+        if self.password:
+            self.password, django_password = self.generate_passwords(self.password)
         mongo_user = super().save()
-        User = get_user_model()
-        django_user = User(username=self.username, email=self.email, password=self.django_password)
-        django_user.save()
+        get_user_model().objects.create_user(username=self.username,
```

Comment:
It is a bad approach to use bare function calll.
Can be changed like
user_model = get_user_model()